prov/efa/src/efa_av.c

```diff
@@ -1075,3 +1075,97 @@ int efa_av_open(struct fid_domain *domain_fid, struct fi_av_attr *attr,
 	free(av);
 	return ret;
 }
+
+#ifdef EFA_UNIT_TEST
+
+#include "efa_unit_tests.h"
+
+/* Common Setup/Teardown */
+
+struct efa_resource {
+	struct fid_fabric *fabric;
+	struct fid_domain *domain;
+	struct fid_av *av;
+};
+
+struct efa_resource *create_efa_resource() {
+    struct efa_resource *resource = (struct efa_resource*) malloc(sizeof(struct efa_resource));
+    resource->fabric = NULL;
+    resource->domain = NULL;
+    resource->av = NULL;
+
+    struct fi_info *info;
+    int rv = fi_getinfo(FI_VERSION(1, 14), NULL, NULL, 0ULL, NULL, &info);
+    if (rv) {
+        free(resource);
+        return NULL;
+    }
+
+    rv = fi_fabric(info->fabric_attr, &resource->fabric, NULL);
+    if (rv) {
+        free(resource);
+        return NULL;
+    }
+
+    rv = fi_domain(resource->fabric, info, &resource->domain, NULL);
+    if (rv) {
+        fi_close((fid_t) resource->fabric);
+        free(resource);
+        return NULL;
+    }
+
+    struct fi_av_attr av_attr = {0};
```

Comment:
Typically we put variable declarations at the top of the function

Reply:
done globally.